algorithms.go

```diff
@@ -390,19 +422,26 @@ func newAlgorithm(algo string, key []byte) (hash.Hash, crypto.Hash, error) {
 func signerFromString(s string) (signer, error) {
 	s = strings.ToLower(s)
 	isEcdsa := false
+	isEd25519 := false
 	var algo string = ""
 	if strings.HasPrefix(s, ecdsaPrefix) {
 		algo = strings.TrimPrefix(s, ecdsaPrefix+"-")
 		isEcdsa = true
 	} else if strings.HasPrefix(s, rsaPrefix) {
 		algo = strings.TrimPrefix(s, rsaPrefix+"-")
+	} else if strings.HasPrefix(s, ed25519Prefix) {
+		isEd25519 = true
+		algo = "sha512"
 	} else {
 		return nil, fmt.Errorf("no signer matching %q", s)
 	}
```

Comment:
Note to self: this is getting to the point where it is not going to scale.

No action required for this comment.